tests/phpunit/Civi/WorkflowMessage/ExampleWorkflowMessageTest.php

```diff
@@ -259,22 +259,52 @@ public function testImpromptuImportExport() {
     $this->assertTrue(!isset($envelope['myProtectedInt']));
   }
 
+  public function testExampleRender() {
+    $hookCount = 0;
+    $rand = rand(0, 1000);
+    $cid = $this->individualCreate(['first_name' => 'Foo', 'last_name' => 'Bar' . $rand, 'prefix_id' => NULL, 'suffix_id' => NULL]);
+    /** @var \Civi\WorkflowMessage\GenericWorkflowMessage $ex */
+    $ex = $this->createExample();
+    \Civi::dispatcher()->addListener('hook_civicrm_alterMailParams', function($e) use (&$hookCount) {
+      $hookCount++;
+      $this->assertEquals('my_example_wf', $e->params['valueName'], 'ExampleWorkflow::WORKFLOW should propagate to params[valueName]');
```

Comment:
I think it would be more correct to say that if valueName is passed in it should get as far as the hook

Reply:
(We should ensure that for new code AND existing code workflow is set and existing code can continue to set valueName as it does)